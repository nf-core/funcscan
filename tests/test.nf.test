nextflow_pipeline {

    name "Test pipeline: NFCORE_FUNCSCAN"
    script "main.nf"
    tag "pipeline"
    tag "nfcore_funcscan"
    tag "test"

    test("test_profile") {

        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/pipeline_info/nf_core_funcscan_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },

                // AMPir
                { assert snapshot(
                    file("$outputDir/amp/ampir/sample_1/sample_1.ampir.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_1/sample_1.ampir.faa").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_2/sample_2.ampir.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/ampir/sample_2/sample_2.ampir.faa").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                ).match("ampir") },

                // AMPlify
                { assert snapshot(
                    file("$outputDir/amp/amplify/sample_1/sample_1.amplify.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1"),
                    file("$outputDir/amp/amplify/sample_2/sample_2.amplify.tsv").text.contains("NODE_882919_length_258_cov_0.935961_1")
                ).match("amplify") },

                // HMMsearch
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_1/sample_1_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },
                { assert new File("$outputDir/amp/hmmer_hmmsearch/sample_2/sample_2_mybacteriocin.hmmer_hmmsearch.txt.gz").exists() },

                // Macrel
                { assert snapshot(
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.smorfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.all_orfs.faa.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel.prediction.gz"),
                    path("$outputDir/amp/macrel/sample_1.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_1.macrel/README.md"),
                    path("$outputDir/amp/macrel/sample_1.macrel/sample_1.macrel_log.txt"),
                    path("$outputDir/amp/macrel/sample_2.macrel/sample_2.macrel_log.txt")
                ).match("macrel") },

                // AMPcombi
                { assert path("$outputDir/reports/ampcombi2/Ampcombi_summary.tsv").text.contains("NODE_515831_length_303_cov_1.532258_1") },
                { assert path("$outputDir/reports/ampcombi2/Ampcombi_parse_tables.log").text.contains("amp_DRAMP_database is found and will be used") },
                { assert path("$outputDir/reports/ampcombi2/Ampcombi_parse_tables.log").text.contains('$$  _$$  _$$') },
                { assert snapshot(
                    path("$outputDir/reports/ampcombi2/Ampcombi_cluster.log"),
                    path("$outputDir/reports/ampcombi2/Ampcombi_complete.log")
                ).match("ampcombi_logfiles") },

                // DeepARG
                { assert snapshot(
                    path("$outputDir/arg/deeparg/sample_1/sample_1.align.daa.tsv"),
                    path("$outputDir/arg/deeparg/sample_2/sample_2.align.daa.tsv"),
                    path("$outputDir/arg/deeparg/sample_1/sample_1.mapping.ARG"),
                    path("$outputDir/arg/deeparg/sample_2/sample_2.mapping.ARG")
                ).match("deeparg_tsv_ARG") },
                { assert file("$outputDir/arg/deeparg/sample_1/sample_1.align.daa").name },
                { assert file("$outputDir/arg/deeparg/sample_2/sample_2.align.daa").name },
                { assert path("$outputDir/arg/deeparg/sample_1/sample_1.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_2/sample_2.mapping.potential.ARG").text.contains("#ARG") },
                { assert path("$outputDir/arg/deeparg/sample_1/sample_1.align.daa.tsv").text.contains("rifampin_monooxygenase|rifamycin|rifampin_monooxygenase") },
                { assert path("$outputDir/arg/deeparg/sample_2/sample_2.align.daa.tsv").text.contains("rifampin_monooxygenase|rifamycin|rifampin_monooxygenase") },

                // ABRicate
                { assert snapshot(
                    path("$outputDir/arg/abricate/sample_1/sample_1.txt"),
                    path("$outputDir/arg/abricate/sample_2/sample_2.txt"),
                ).match("abricate") },

                // AMRFinderPlus
                { assert snapshot(
                    path("$outputDir/arg/amrfinderplus/sample_1/sample_1.tsv"),
                    path("$outputDir/arg/amrfinderplus/sample_2/sample_2.tsv"),
                ).match("amrfinderplus") },

                // RGI
                { assert snapshot(
                    path("$outputDir/arg/rgi/sample_1/sample_1.txt"),
                    path("$outputDir/arg/rgi/sample_2/sample_2.txt"),
                ).match("rgi") },

                // fARGene
                { assert snapshot(
                    path("$outputDir/arg/fargene/sample_1/class_a/results_summary.txt"),
                    path("$outputDir/arg/fargene/sample_2/class_a/results_summary.txt"),
                    path("$outputDir/arg/fargene/sample_1/class_b_1_2/results_summary.txt"),
                    path("$outputDir/arg/fargene/sample_2/class_b_1_2/results_summary.txt")
                ).match("fargene")
                },
                { assert path("$outputDir/arg/fargene/sample_1/fargene_analysis.log").text.contains("fARGene is done.") },
                { assert path("$outputDir/arg/fargene/sample_2/fargene_analysis.log").text.contains("fARGene is done.") },

                // hAMRonization
                { assert snapshot(path("$outputDir/reports/hamronization_summarize/hamronization_combined_report.tsv")).match("hamronization_summarize") }
            )
        }
    }
}
