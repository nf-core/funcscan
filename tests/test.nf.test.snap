{
    "hamronization_summarize": {
        "content": [
            "hamronization_combined_report.tsv:md5,864466b0fb1acfc0e6b3425271f78ecb"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T15:33:45.641163055"
    },
    "abricate": {
        "content": [
            "sample_1.txt:md5,69af3321b0bc808b7ef85f102395736f",
            "sample_2.txt:md5,69af3321b0bc808b7ef85f102395736f"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T15:33:45.590739146"
    },
    "fargene": {
        "content": [
            "results_summary.txt:md5,690d351cfc52577263ef4cfab1c81f50",
            "results_summary.txt:md5,690d351cfc52577263ef4cfab1c81f50",
            "results_summary.txt:md5,f69af9b0e18a0cfc934eb18b7e4bffe1",
            "results_summary.txt:md5,f69af9b0e18a0cfc934eb18b7e4bffe1"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T15:33:45.625398198"
    },
    "rgi": {
        "content": [
            "sample_1.txt:md5,fce130af51f93cccfc09ddaf9caf623f",
            "sample_2.txt:md5,fce130af51f93cccfc09ddaf9caf623f"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T15:33:45.614473482"
    },
    "deeparg_tsv_ARG": {
        "content": [
            "sample_1.align.daa.tsv:md5,21822364379fe8f991d27cdb52a33d1d",
            "sample_2.align.daa.tsv:md5,d59287f357de198639bdca5dbaede173",
            "sample_1.mapping.ARG:md5,0e049e99eab4c55666062df21707d5b9",
            "sample_2.mapping.ARG:md5,0e049e99eab4c55666062df21707d5b9"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T12:41:33.325286058"
    },
    "ampir": {
        "content": [
            true,
            true,
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T12:41:33.055416682"
    },
    "ampcombi_logfiles": {
        "content": [
            "Ampcombi_cluster.log:md5,4c78f5f134edf566f39e04e3ab7d8558",
            "Ampcombi_complete.log:md5,3dabfea4303bf94bd4f5d78c5b8c83c1"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T12:41:33.230701016"
    },
    "amplify": {
        "content": [
            true,
            true
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T12:41:33.1312123"
    },
    "macrel": {
        "content": [
            "sample_1.macrel.smorfs.faa.gz:md5,1b5e2434860e635e95324d1804a3be7b",
            "sample_2.macrel.smorfs.faa.gz:md5,38108b5cdfdc2196afe67418b9b04682",
            "sample_1.macrel.all_orfs.faa.gz:md5,844bb10e2f84e1a2b2db56eb36391dcf",
            "sample_2.macrel.all_orfs.faa.gz:md5,9c0b8b1c3b03d7b20aee0b57103861ab",
            "sample_1.macrel.prediction.gz:md5,9553e1dae8a5b912da8d74fa3f1cd9eb",
            "sample_2.macrel.prediction.gz:md5,ae155e454eb7abd7c48c06aad9261603",
            "README.md:md5,cf088d9256ff7b7730699f17b64b4028",
            "README.md:md5,cf088d9256ff7b7730699f17b64b4028",
            "sample_1.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
            "sample_2.macrel_log.txt:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-18T10:35:54.749106433"
    },
    "amrfinderplus": {
        "content": [
            "sample_1.tsv:md5,b4d261ace9be7d013c19d1f5c0005bfe",
            "sample_2.tsv:md5,b4d261ace9be7d013c19d1f5c0005bfe"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.3"
        },
        "timestamp": "2024-07-23T15:33:45.603392278"
    }
}